shared_models/shared_models.py

```diff
@@ -100,13 +103,54 @@ class Endorsement(BaseModel):
     state: str
 
 
+class ServiceDecorator(TypedDict):
+    endpoint: str
+    recipient_keys: List[str]
+    routing_keys: Optional[List[str]]
+
+
+class Oob(BaseModel):
```

Comment:
It would be good to use literal/enum/union types as much as possible (e.g. for state/role etc...)